in_toto/models/metadata.py

```diff
@@ -79,38 +254,9 @@ def __repr__(self):
       )
 
 
-  def dump(self, path):
```

Comment:
Speaking of `compact_json`, I found, it was actually never used.
It was only referenced in `in_toto_run` at https://github.com/in-toto/in-toto/blob/fbc1eb053dd39c6d1fb307a60bd2c781e667c931/in_toto/runlib.py#L658-L662

which again is always `False` because no command line argument is provided to use this option.
https://github.com/in-toto/in-toto/blob/fbc1eb053dd39c6d1fb307a60bd2c781e667c931/in_toto/runlib.py#L512-L516
https://github.com/in-toto/in-toto/blob/fbc1eb053dd39c6d1fb307a60bd2c781e667c931/in_toto/in_toto_run.py#L221-L227

Reply:
The new `Metadata.dump()` method takes care of compact json using optional argument `compact` that I forgot to mention in doc-string.